### PR TITLE
Alter login flow to allow for email auth login

### PR DESCRIPTION
**Pull request by gov-cjwaszczuk**
# Summary
Added the login functionality for a user currently on email auth. If they log in correctly, they will be sent an email with a token inside it (with 'next' query string appended to it if they came to the log-in from a different notify request). The user will then be able to click the link to log into the platform. 

## :o: Two factor email sent 
image
## :o: Resend email 
image
## :o: Two factor email
image

### CODE REVIEW

File: app/templates/views/email-not-received.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  Resend email link
+{% endblock %}
+
+{% block maincolumn_content %}
+
+<div class="grid-row">
+  <div class="column-two-thirds">
+    <h1 class="heading-large">Resend email link</h1>
+
+    <p> Email messages sometimes take a few minutes to arrive. If you do not reiceve the email, you can resend it.</p>
```
Comment (klssmith) [inline app/templates/views/email-not-received.html]: Small typo here (reiceve)

File: app/templates/views/email-not-received.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{% extends "withoutnav_template.html" %}
+{% from "components/textbox.html" import textbox %}
```
Comment (klssmith) [inline app/templates/views/email-not-received.html]: Do we need to import the textbox, checkbox and page-footer for this page?

File: app/templates/views/two-factor-email.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+{% block per_page_title %}
+  Email verification
+{% endblock %}
+
+{% block maincolumn_content %}
+
+<div class="grid-row">
+  <div class="column-two-thirds">
+    <h1 class="heading-large">{{ title }}</h1>
+    <p> We've sent you an email with your login link</p>
```
Comment (quis) [inline app/templates/views/two-factor-email.html]: Should be a curly apostrophe, ie `We’ve` or `alt` + `⇧` + `]`

File: tests/app/main/views/test_two_factor.py [hunk 4]
```diff
@@ -213,3 +215,108 @@ def test_two_factor_should_activate_pending_user(
...
+            url_for('main.two_factor_email', token=valid_token),
+            follow_redirects=True,
+        )
+
+    assert response.status_code == 200
+    page = BeautifulSoup(response.data.decode('utf-8'), 'html.parser')
+
+    assert normalize_spaces(
+        page.select_one('.banner-dangerous').text
+    ) == "The link in the email we sent you has expired. We've sent you a new one."
```
Comment (quis) [inline tests/app/main/views/test_two_factor.py]: Proper apostrophe again.

File: tests/app/main/views/test_two_factor.py [hunk 5]
```diff
@@ -213,3 +215,108 @@ def test_two_factor_should_activate_pending_user(
...
+
+    response = client.get(
+        url_for('main.two_factor_email', token=valid_token),
+        follow_redirects=True
+    )
+
+    page = BeautifulSoup(response.data.decode('utf-8'), 'html.parser')
+    assert normalize_spaces(
+        page.select_one('.banner-dangerous').text
+    ) == "There's something wrong with the code"
```
Comment (quis) [inline tests/app/main/views/test_two_factor.py]: Again 😛

File: app/main/views/two_factor.py [hunk 6]
```diff
@@ -1,16 +1,64 @@
...
+    except SignatureExpired as exc:
+        # lets decode again, without the expiry, to get the user id out
+        orig_data = json.loads(check_token(
+            token,
+            current_app.config['SECRET_KEY'],
+            current_app.config['DANGEROUS_SALT'],
+            None
+        ))
+        session['user_details'] = {'id': orig_data['user_id']}
+        flash("The link in the email we sent you has expired. We've sent you a new one.")
```
Comment (quis) [inline app/main/views/two_factor.py]: Apostrophe.

**Pull request merged**